post: add go-ipfs 0.7 post

jacobheun opened the pull request:
~~Pending final release of 0.7, https://github.com/ipfs/go-ipfs/issues/7560~~

It's ready!

aschmahmann commented:
@jacobheun should be updated based on the latest release notes (i.e. noting the build change)

jacobheun commented:
The latest changes to the release notes have been added. I also added in links to the changelog, updated the total contributor count for this release and added an Intro focused on SECIO and noting the Peer ID change.

lidel commented on content/post/110-go-ipfs-0-7-0.md:
```diff
@@ -0,0 +1,80 @@
+
+## #️⃣ IPNS paths now encode the key name as a base36 CIDv1 by default
+
+Previously go-ipfs encoded the key names for IPNS paths as base58btc multihashes (e.g. Qmabc...). We now encode them as base36 encoded CIDv1s as defined in the [peerID spec](https://github.com/libp2p/specs/blob/master/peer-ids/peer-ids.md#string-representation) (e.g. k51xyz...) which also deals with the encoding of public keys. This is nice because it means that IPNS keys will by default be case- ...
```
nit: I find it easier to read when cids and hostnames are formatted as `Qm..` 

```suggestion
Previously go-ipfs encoded the key names for IPNS paths as base58btc multihashes (e.g. `Qmabc...`). We now encode them as base36 encoded CIDv1s as defined in the [peerID spec](https://github.com/libp2p/specs/blob/master/peer-ids/peer-ids.md#string-representation) (e.g. `k51xyz...`) which also deals with the encoding of public keys. This is nice because it means that IPNS keys will by default be case-insensitive and that they will fit into DNS labels (e.g. `k51xyz...ipns.localhost`) and therefore that subdomain gateway redirections (e.g. from `localhost:8080/ipns/{key}` to `{key}.ipns.localhost`) will look better to users in the default case.
```

lidel commented on content/post/110-go-ipfs-0-7-0.md:
```diff
@@ -0,0 +1,80 @@
+
+Previously go-ipfs encoded the key names for IPNS paths as base58btc multihashes (e.g. Qmabc...). We now encode them as base36 encoded CIDv1s as defined in the [peerID spec](https://github.com/libp2p/specs/blob/master/peer-ids/peer-ids.md#string-representation) (e.g. k51xyz...) which also deals with the encoding of public keys. This is nice because it means that IPNS keys will by default be case- ...
+
+Many commands will accept a `--ipns-base` option that allows changing command outputs to use a particular encoding (i.e.  base58btc multihash, or CIDv1 encoded in any supported base)
```
Maybe we could add some examples?

````suggestion
Many commands will accept a `--ipns-base` option that allows changing command outputs to use a particular encoding (i.e.  base58btc multihash, or CIDv1 encoded in any supported base):

```console
$ ipfs key list -l --ipns-base b58mh
12D3KooWCjhz69LskTZEC5vFWs8eDpHo7kYbGzrC5EjU75BHSmVK self

$ ipfs key list -l --ipns-base base36
k51qzi5uqu5dh9ihj4p2v5sl3hxvv27ryx2w0xrsv6jmmqi91t9xp8p9kaipc2 self
```
````

lidel commented on content/post/110-go-ipfs-0-7-0.md:
```diff
@@ -0,0 +1,80 @@
+
+## 📮  Multiaddresses now accept PeerIDs encoded as CIDv1
+
+In preparation for eventually changing the default PeerID representation multiaddresses can now contain strings like `/p2p/k51xyz...` in addition to the default `/p2p/Qmabc...`. There is a corresponding `--peerid-base` option to many functions that output peerIDs.
```
As before, adding example:

````suggestion
In preparation for eventually changing the default PeerID representation multiaddresses can now contain strings like `/p2p/k51xyz...` in addition to the default `/p2p/Qmabc...`. There is a corresponding `--peerid-base` option to many functions that output peerIDs: 

```console
$ ipfs id --format "<id>" --peerid-base b58mh
12D3KooWCjhz69LskTZEC5vFWs8eDpHo7kYbGzrC5EjU75BHSmVK

$ ipfs id --format "<id>" --peerid-base base36
k51qzi5uqu5dh9ihj4p2v5sl3hxvv27ryx2w0xrsv6jmmqi91t9xp8p9kaipc2
```
````

lidel commented on content/post/110-go-ipfs-0-7-0.md:
````diff
@@ -0,0 +1,80 @@
+
+Initial support has been added for the `ipfs dag stat` command. Running this command will traverse the DAG for the given root CID and report statistics. By default, progress will be shown as the DAG is traversed. Supported statistics currently include DAG size and number of blocks.
+
+```bash
+ipfs dag stat bafybeihpetclqvwb4qnmumvcn7nh4pxrtugrlpw4jgjpqicdxsv7opdm6e # the IPFS webui
+Size: 30362191, NumBlocks: 346
+```
````
nit: when snippet includes both input and output, including `$` makes it easier to tell which line was the command:

````suggestion
```bash
$ ipfs dag stat bafybeihpetclqvwb4qnmumvcn7nh4pxrtugrlpw4jgjpqicdxsv7opdm6e # the IPFS webui
Size: 30362191, NumBlocks: 346
```
````

lidel commented on content/post/110-go-ipfs-0-7-0.md:
```diff
@@ -0,0 +1,80 @@
+
+## 🔌 Plugin build changes
+
+We have changed the build flags used by the official binary distributions on dist.ipfs.io (or `/ipns/dist.ipfs.io`) to use the simpler and more reliable `-trimpath` flag instead of the more complicated and brittle `-asmflags=all=-trimpath="$(GOPATH)" -gcflags=all=-trimpath="$(GOPATH)"` flags, however the build flags used by default in go-ipfs remain the same.
```
```suggestion
We have changed the build flags used by the official binary distributions on [dist.ipfs.io](https://dist.ipfs.io) (or `/ipns/dist.ipfs.io`) to use the simpler and more reliable `-trimpath` flag instead of the more complicated and brittle `-asmflags=all=-trimpath="$(GOPATH)" -gcflags=all=-trimpath="$(GOPATH)"` flags, however the build flags used by default in go-ipfs remain the same.
```

lidel commented on content/post/110-go-ipfs-0-7-0.md:
```diff
@@ -0,0 +1,80 @@
+
+We have changed the build flags used by the official binary distributions on dist.ipfs.io (or `/ipns/dist.ipfs.io`) to use the simpler and more reliable `-trimpath` flag instead of the more complicated and brittle `-asmflags=all=-trimpath="$(GOPATH)" -gcflags=all=-trimpath="$(GOPATH)"` flags, however the build flags used by default in go-ipfs remain the same.
+
+The scripts in https://github.com/ipfs/go-ipfs-example-plugin have been updated to reflect this change. This is a breaking change to how people have been building plugins against the dist.ipfs.io binary of go-ipfs and plugins should update their build processes accordingly see https://github.com/ipfs/go-ipfs-example-plugin/pull/9 for details.
```
```suggestion
The scripts in [go-ipfs-example-plugin](https://github.com/ipfs/go-ipfs-example-plugin) have been updated to reflect this change. This is a breaking change to how people have been building plugins against the dist.ipfs.io binary of go-ipfs and plugins should update their build processes accordingly. See [go-ipfs-example-plugin/pull/9](https://github.com/ipfs/go-ipfs-example-plugin/pull/9) for details.
```

lidel commented on content/post/110-go-ipfs-0-7-0.md:
```diff
@@ -0,0 +1,80 @@
+
+## Install, upgrade, and join us!
+
+You can get started by [installing go-ipfs](https://dist.ipfs.io/#go-ipfs) or [upgrading to go-ipfs 0.7](https://docs.ipfs.io/recent-releases/go-ipfs-0-7/update-procedure).
```
https://docs.ipfs.io/recent-releases/go-ipfs-0-7/update-procedure is missing atm –  @johnnymatthews  is there a PR for adding it?

johnnymatthews replied:
There isn't a page for `recent-releases/go-ipfs-0-7` yet. I'll make it now though. Gonna steal heavily from this blog post. Hope that's ok.

jacobheun replied:
I'll create a PR to the ipfs repo to include the update process docs in the release check list so we can make sure we start that update process earlier.

johnnymatthews replied:
Sorry for the delay. [docs.ipfs.io/recent-releases/go-ipfs-0-7-0](https://docs.ipfs.io/recent-releases/go-ipfs-0-7/) is now live-ish. I just pushed, so it might take about an hour or so to propagate through the interwebs.

lidel reviewed (approved):
LGTM, added some links and examples, but feel free to ignore those suggestions and merge as-is.

aschmahmann commented on content/post/110-go-ipfs-0-7-0.md:
```diff
@@ -0,0 +1,80 @@
+
+In August we announced the [deprecation of the SECIO security transport](https://blog.ipfs.io/2020-08-07-deprecating-secio/). In this release we have disabled SECIO by default, which will have an impact on older nodes on the network. The best way to mitigate the impact of this change is to [upgrade your IPFS nodes](https://docs.ipfs.io/recent-releases/go-ipfs-0-7/update-procedure) as soon as poss ...
+
+With this release you will also start seeing more Peer IDs on the network that start with `1` instead of the typical `Qm`. This is due to a switch to ed25519 keys being used by default over RSA keys, which you can read more about in the highlights below.
```
> more Peer IDs on the network

also more IPNS keys, or not worth highlighting?

aschmahmann commented on content/post/110-go-ipfs-0-7-0.md:
````diff
@@ -0,0 +1,80 @@
+Size: 30362191, NumBlocks: 346
+```
+
+## 🔌 Plugin build changes
````
Wondering if we should add an 🚨 or something here to help highlight it's a breaking change for some of the plugin folks. Perhaps it doesn't really matter though since you're either reading the highlights or you've already skipped them and it's too late.

jacobheun replied:
I added the alert icons around it and also added a note in the intro with an alert to check out these changes for folks using plugins.

aschmahmann reviewed (approved):
LGTM, left a couple nits that you can feel free to ignore if you don't think they're relevant.
